### PR TITLE
modify logic in QScintilla easyblock to find the PyQt5 sipdir in more places

### PR DESCRIPTION
**Pull request by mboisson**
When installing PyQt5 5.15.5 as an extension of Qt5 5.15.8, two things happen
1) The current `qscintilla.py` at line https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/q/qscintilla.py#L127 tests `get_software_version('PyQt5')`, which returns None, since `PyQt5` is it is part of `Qt5`, not `PyQt5` module (see support for this at https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/q/qscintilla.py#L60)

2) Our SIP files are not in the locations that are tested, i.e. previously, we had: 
```
$ find /cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.12.8/ -name 'qgraphicssceneevent.sip'
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.12.8/share/python3.6/site-packages/sip/PyQt5/QtWidgets/qgraphicssceneevent.sip
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.12.8/share/python3.7/site-packages/sip/PyQt5/QtWidgets/qgraphicssceneevent.sip
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.12.8/share/python3.8/site-packages/sip/PyQt5/QtWidgets/qgraphicssceneevent.sip
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.12.8/share/python3.9/site-packages/sip/PyQt5/QtWidgets/qgraphicssceneevent.sip
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.12.8/share/python2.7/site-packages/sip/PyQt5/QtWidgets/qgraphicssceneevent.sip
``` 

now we have: 
```
$ find /cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.15.8/ -name 'qgraphicssceneevent.sip'
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.15.8/lib/python3.8/site-packages/PyQt5/bindings/QtWidgets/qgraphicssceneevent.sip
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.15.8/lib/python3.9/site-packages/PyQt5/bindings/QtWidgets/qgraphicssceneevent.sip
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qt/5.15.8/lib/python3.10/site-packages/PyQt5/bindings/QtWidgets/qgraphicssceneevent.sip
``` 

This PR avoids using `get_software_version('PyQt5')` altogether, and instead tests various possible location of PyQt5 sip dir, and picks the first one that matches.

### CODE REVIEW

File: easybuild/easyblocks/q/qscintilla.py [hunk 1]
```diff
@@ -118,16 +118,23 @@ def install_step(self):
...
 
             sip_incdir = find_glob_pattern(os.path.join(self.pyqt_root, 'include', 'python%s*' % pyshortver), False)
-            # in case PyQt5's sip was installed in directories that are specific to each version of python
-            # as could happen with multi_deps
-            pyqt_sipdir = find_glob_pattern(os.path.join(self.pyqt_root, 'share', 'python%s*' % pyshortver,
-                                                         'site-packages', 'sip', self.pyqt_pkg_name), False)
-            # fall back to a single sipdir
+            # depending on PyQt5 versions and how it was installed, the sip directory could be in various places
+            # test them and figure out the first one that matches
+            pyqt_sip_subidr = [os.path.join('share', 'python%s*' % pyshortver, 'site-packages', 'sip',
```
Comment (Micket) [inline easybuild/easyblocks/q/qscintilla.py:123]:
```suggestion
            pyqt_sip_subdIr = [os.path.join('share', 'python%s*' % pyshortver, 'site-packages', 'sip',
```

File: easybuild/easyblocks/q/qscintilla.py [hunk 2]
```diff
@@ -118,16 +118,23 @@ def install_step(self):
...
+                                            self.pyqt_pkg_name),
+                               os.path.join('share', 'sip', self.pyqt_pkg_name),
+                               os.path.join('share', 'sip'),
+                               os.path.join('lib', 'python%s*' % pyshortver, 'site-packages', self.pyqt_pkg_name,
+                                            'bindings')
+                               ]
+            pyqt_sipdir_options = [os.path.join(self.pyqt_root, subdir) for subdir in pyqt_sip_subidr]
```
Comment (Micket) [inline easybuild/easyblocks/q/qscintilla.py:130]:
```suggestion
            pyqt_sipdir_options = [os.path.join(self.pyqt_root, subdir) for subdir in pyqt_sip_subdir]
```

**Review by Micket — CHANGES_REQUESTED** [review]
Apart from the typo, looks good i think. But you won't have pyqt_root as well right? Since you don't have PyQt5 as an extension.

**Review by Micket — APPROVED** [review]
lgtm (i need to find time to make a test build)

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _QScintilla-2.11.6-GCCcore-11.2.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3104.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/606cc6b852159f0d9f567b980b5fd0a0 for a full test report.

**Pull request merged**